APCAppCore/APCAppCore/Startup/APCAppDelegate.m

```diff
@@ -861,43 +869,46 @@ - (NSDictionary *) tasksAndSchedulesWillBeLoaded {
+        for (HKQuantitySample *quantitySample in quantitySamples) {
+            NSString *healthKitType = nil;
+            NSString *quantityValue = nil;
+            NSString *dateTimeStamp = [quantitySample.startDate toStringInISO8601Format];
             
-            // Get the difference in seconds between the start and end date for the sample
-            NSDateComponents *secondsSpentInBedOrAsleep = [[NSCalendar currentCalendar] components:NSCalendarUnitSecond
-                                                                                          fromDate:catSample.startDate
-                                                                                            toDate:catSample.endDate
-                                                                                           options:NSCalendarWrapComponents];
-            if (catSample.value == HKCategoryValueSleepAnalysisInBed) {
-                quantityValue = [NSString stringWithFormat:@"%ld,seconds in bed", (long)secondsSpentInBedOrAsleep.second];
-            } else if (catSample.value == HKCategoryValueSleepAnalysisAsleep) {
-                quantityValue = [NSString stringWithFormat:@"%ld,seconds asleep", (long)secondsSpentInBedOrAsleep.second];
+            if ([quantitySample isKindOfClass:[HKCategorySample class]]) {
+                HKCategorySample *catSample = (HKCategorySample *)quantitySample;
+                healthKitType = catSample.categoryType.identifier;
+                quantityValue = [NSString stringWithFormat:@"%ld", (long)catSample.value];
+                
+                // Get the difference in seconds between the start and end date for the sample
+                NSDateComponents *secondsSpentInBedOrAsleep = [[NSCalendar currentCalendar] components:NSCalendarUnitSecond
+                                                                                              fromDate:catSample.startDate
+                                                                                                toDate:catSample.endDate
+                                                                                               options:NSCalendarWrapComponents];
+                if (catSample.value == HKCategoryValueSleepAnalysisInBed) {
+                    quantityValue = [NSString stringWithFormat:@"%ld,seconds in bed", (long)secondsSpentInBedOrAsleep.second];
+                } else if (catSample.value == HKCategoryValueSleepAnalysisAsleep) {
+                    quantityValue = [NSString stringWithFormat:@"%ld,seconds asleep", (long)secondsSpentInBedOrAsleep.second];
+                }
+            } else {
+                HKQuantitySample *qtySample = (HKQuantitySample *)quantitySample;
+                healthKitType = qtySample.quantityType.identifier;
+                
+                quantityValue = [NSString stringWithFormat:@"%@", qtySample.quantity];
+                quantityValue = [quantityValue stringByReplacingOccurrencesOfString:@" " withString:@","];
```

Comment:
Good point. Should I include both the bundleIdentifier and name for the source or either one? I think bundleIdentifier should be fine since that will be same, even if the name of the app (for whatever reason) changes.